lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php

```diff
@@ -816,6 +822,24 @@ public function generateVersionNumber(\DateTimeInterface $now=null)
     }
 
     /**
+     * Explicitely opens the database connection. This is done to play nice
+     * with DBAL's MasterSlaveConnection. Which, in some cases, connects to a
+     * follower when fetching the executed migrations. If a follower is lagging
+     * significantly behind that means the migrations system may see unexecuted
+     * migrations that were actually executed earlier.
+     *
+     * @return bool The same value returned from the `connect` method
+     */
+    protected function connect()
+    {
+        if ($this->connection instanceof MasterSlaveConnection) {
+            return $this->connection->connect('master');
```

Comment:
@chrisguitarguy This does assume that there is always a connection named master in a master slave connection setup right ?
If so, is it the case or should we make the name of the master connection configurable ?